### ISSUE TITLE
Panic "index out of bounds"

### ISSUE THREAD

**Issue by pftbest** [nsforth/nmea0183#1]
```Rust
fn main() {
    let line = "\u{3}$GPG\u{3}GA,1\u{3}5502\u{3}2.17\u{3}5,53\u{3}50.2\u{3}328,\u{3}N,02\u{3}728.\u{3}5283\u{3},E,1\u{3},5,1\u{2}.4\u{3}0,24\u{3}1.3,\u{3}M,26\u{3}.4,M\u{3},,*5\u{3}1\r\n";

    for result in Parser::new().parse_from_bytes(line.as_bytes()) {
        println!("{:?}", result);
    }
}
```

> thread 'main' panicked at 'index out of bounds: the len is 79 but the index is 79', /Users/vadzim/.cargo/registry/src/github.com-1ecc6299db9ec823/nmea0183-0.2.1/src/lib.rs:355:21

**Comment by nsforth** [comment]
Thank's for reporting and reproducer. I''l check it soon.

**Comment by nsforth** [comment]
Fixed in 0.2.2.
